src/main/java/eu/dissco/backend/service/DigitalMediaService.java

```diff
@@ -141,25 +127,24 @@ private JsonApiListResponseWrapper wrapResponse(List<JsonApiData> dataNodePlusOn
 
   public JsonApiListResponseWrapper getMass(String id, String path) {
     var digitalMedia = repository.getLatestDigitalMediaObjectById(id);
-    var digitalSpecimen = specimenRepository.getLatestSpecimenById(getDsDoiFromDmo(digitalMedia));
+    var digitalSpecimen = digitalSpecimenRepository.getLatestSpecimenById(getDsDoiFromDmo(digitalMedia));
     var flattenObjectData = flattenAttributes(digitalMedia, digitalSpecimen);
     return masService.getMassForObject(flattenObjectData, path);
   }
 
-  private String getDsDoiFromDmo(DigitalMediaObjectWrapper digitalMedia) {
-    for (var entityRelationship : digitalMedia.digitalEntity()
-        .getEntityRelationships()) {
-      if (entityRelationship.getEntityRelationshipType().equals("hasDigitalSpecimen")) {
-        return entityRelationship.getObjectEntityIri().replace(DOI_STRING, "");
+  private String getDsDoiFromDmo(DigitalMedia digitalMedia) {
+    for (eu.dissco.backend.schema.EntityRelationship entityRelationship : digitalMedia.getOdsHasEntityRelationship()) {
```

Comment:
This would be a good place for `var` 
` for (var entityRelationship : digitalMedia.getOdsHasEntityRelationship())`

Reply:
Much better